enhancements/cvo/cluster-profiles.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Cluster Profiles
+authors:
+  - "@csrwng"
+reviewers:
+  - "@abhinavdahiya"
+  - "@crawford"
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-02-04
+last-updated: 2019-02-04
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# cluster-profiles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Cluster profiles are a way to support different deployment models for OpenShift clusters. 
+A profile is an identifier that the Cluster Version Operator uses to determine
+which manifests to apply. Operators can be excluded completely or can have different
+manifests for each supported profile.
+
+## Motivation
+
+To support different a deployment model in which not all operators rendered by
+the CVO by default are needed. This includes IBM Public Cloud, in which a 
+hosted control plane is used. Potentially it can also be used for Code Ready Containers.
+
+### Goals
+
+- Equip the CVO to include/exclude manifests based on a matching identifier (profile)
+
+### Non-Goals
+
+- Define which manifests should be excluded/included for each use case
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a user, I can create a cluster in which manifests for control plane operators are
+not applied by the CVO.
+
+#### Story 2
+As a user, I can create a cluster in which node selectors for certain operators target
+worker nodes instead of master nodes.
+
+### Design
+
+A cluster profile is specified to the CVO as an identifier in an environment
+variable. For a given cluster, only one CVO profile may be in effect.
+
+NOTE: The mechanism by which the environment variable is set on the CVO deployment is 
+out of the scope of this design.
```

Comment:
Are there notes about how this works for IBM today?  Are the CRC folks on-board with managing their own CVO Deployment (seems like that might make updates difficult, as the CVO updates its own Deployment to bump itself?)?

Reply:
We (CRC) were actually considering using a config map to set the profile, but I don't know how well it fits the IBM use case.
See https://github.com/openshift/cluster-version-operator/pull/404/files#diff-d540a41404f678a1c438f4c1e5b92a87R330-R337